src/main/java/bussiness_layer/services/ProfessorService.java

```diff
@@ -1,40 +1,63 @@
 package bussiness_layer.services;
 
 import bussiness_layer.dto.EnrollmentDTO;
-import bussiness_layer.dto.GroupDTO;
 import bussiness_layer.dto.TeachingDTO;
-import data_layer.domain.Enrollment;
-import data_layer.domain.Teaching;
-import data_layer.repositories.IEnrollmentRepository;
-import data_layer.repositories.ITeachingRepository;
+import data_layer.domain.*;
+import data_layer.repositories.*;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import utils.TeachingHandler;
+import utils.exceptions.AccessForbiddenException;
+import utils.exceptions.ResourceNotFoundException;
 
 import javax.transaction.Transactional;
 import java.util.List;
-import java.util.SortedSet;
 import java.util.stream.Collectors;
 
 @Service
 @Transactional
 public class ProfessorService implements IProfessorService {
```

Comment:
Maybe it will be a better approach to break down this service into professor service and teaching service, otherwise there are too much informations to deal with onto this service.